src/js/jobs/components/JobArgs.tsx

```diff
@@ -0,0 +1,191 @@
+    return (
+        <>
+            <JobArgsRow title="Reference">
+                <Link to={`/refs/${ref_id}`}>{ref_id}</Link>
+            </JobArgsRow>
+            <JobArgsRow title="Index">
+                <Link to={`/refs/${ref_id}/indexes/${index_id}`}>{index_id}</Link>
+            </JobArgsRow>
+        </>
+    );
+}
+
+/**
+ * Rows showing important arguments when running an "create_sample" workflow.
+ *
+ * @param args - The complete list of arguments used to run the job
+ * @returns Table rows containing the arguments of the job
+ */
+export function CreateSampleRows({ sample_id }: { sample_id: string }) {
+    return (
+        <JobArgsRow title="Sample">
+            <Link to={`/samples/${sample_id}`}>{sample_id}</Link>
+        </JobArgsRow>
+    );
+}
+
+/**
+ * Rows showing important arguments when running an "create_subtraction" workflow.
+ *
+ * @param args - The complete list of arguments used to run the job
+ * @returns Table rows containing the arguments of the job
+ */
+export function CreateSubtractionRows({ subtraction_id }: { subtraction_id: string }) {
+    return (
+        <JobArgsRow title="Subtraction">
+            <Link to={`/subtractions/${subtraction_id}`}>{subtraction_id}</Link>
+        </JobArgsRow>
+    );
+}
+
+/**
+ * Rows showing important arguments when running an "update_sample" workflow.
+ *
+ * @param args - The complete list of arguments used to run the job
+ * @returns Table rows containing the arguments of the job
+ */
+export function UpdateSampleRows({ sample_id }: { sample_id: string }) {
+    return (
+        <JobArgsRow title="Subtraction">
+            <Link to={`/samples/${sample_id}`}>{sample_id}</Link>
+        </JobArgsRow>
+    );
+}
+
+const UnsupportedJobArgsRow = styled(JobArgsRow)`
+    th {
+        font-weight: normal;
+        font-family: monospace;
+    }
+`;
+
+/**
+ * Generic rows displaying the arguments passed to the job when the workflow type is not know.
+ *
+ * @param args - The complete list of arguments used to run the job
+ * @returns Table rows containing the arguments of the job
+ */
+function GenericJobArgsRows({ args }: { args: object }) {
+    return (
+        <>
+            {map(args, (value, key): ReactNode => {
+                if (typeof value === "string" || typeof value === "number") {
+                    return (
+                        <UnsupportedJobArgsRow key={key} title={key}>
+                            {value}
+                        </UnsupportedJobArgsRow>
+                    );
+                }
+                return null;
+            })}
+        </>
+    );
+}
+
+/**
+ * The tabel rows containing arguments used to run a job.
+ *
+ * @param workflow - The workflow name
+ * @param args - The complete list of arguments used to run the job
+ * @returns Table rows containing arguments used to run a job
+ */
+export function JobArgsRows({ workflow, args }) {
```

Comment:
I am not sure why some of these components are exported. Testing?

Reply:
Yeah, testing updated to RTL, and exports removed